src/mockServiceWorker.js

```diff
@@ -97,9 +97,11 @@ self.addEventListener('fetch', function (event) {
     return
   }
 
+  const clientPromise = event.target.clients.get(clientId)
```

Comment:
Because we don't have a way to test it, we can add a comment referring to the issue to remember in the future why this line has moved outside. What do you think?

Reply:
I think that reason can be looked up in Git commit history. Now I’m wondering if we can test this somehow... the issue happened only on Safari.

Reply:
Yes I know and I think that we can't run puppeteer in Safari. I know that it can be found in git history but because we don't have a test for it, in case of change, it will not break anything. Of course, if we find a way to test it comment the line will be not very helpful

Reply:
I'll try to analyze what exactly goes different in Safari and see if we can simulate that scenario in a test.

Reply:
I was unable to create a test suite for this. `event.target` is `null` in Safari only, I don't see how I can simulate that in a test. I've tried issuing a request from ClientA, then closing that client, and observing what happens in ClientB controlled by the same worker. I'd expect that since the client was closed, the `event.target` or at least `evet.target.clients.get` will be `null`, but it's not. At least, it doesn't throw an exception.

Reply:
Yes, this morning I have tried to take a look at the issue but no found a way to test too. The only thing that I Have found is that the first time target is not null. It will become `null` after the first `await event.target.clients.get(clientId)`